op-acceptance-tests/tests/base/reth_geth_divergence_test.go

```diff
@@ -0,0 +1,204 @@
+package base
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/ethereum-optimism/optimism/devnet-sdk/system"
+	"github.com/ethereum-optimism/optimism/devnet-sdk/testing/systest"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+	"github.com/ethereum/go-ethereum/ethclient"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/stretchr/testify/require"
+)
+
+// TestRethGethDivergence checks that reth and geth do not diverge in terms of state root.
+// geth is considered the canonical/correct implementation.
+func TestRethGethDivergence(t *testing.T) {
+	systest.SystemTest(t, rethGethDivergenceTestScenario())
+}
+
+func rethGethDivergenceTestScenario() systest.SystemTestFunc {
+	const (
+		// testTimeout is the maximum time allowed for the test to run
+		testTimeout = 60 * time.Second
+
+		// blockCheckCount is the number of recent blocks to check for divergence
+		blockCheckCount = 5
+	)
+
+	return func(t systest.T, sys system.System) {
+		logger := testlog.Logger(t, log.LevelInfo)
+		logger.Info("Started reth-geth divergence test")
+
+		// Check all L2 chains
+		for i, chain := range sys.L2s() {
+			chainIndex := i
+			currentChain := chain
+			t.Run(fmt.Sprintf("Chain_%d", chainIndex), func(t systest.T) {
+				t.Parallel()
+				checkChainForDivergence(t, currentChain, logger.New("chain", chainIndex), testTimeout, blockCheckCount)
+			})
+		}
+	}
+}
+
+// checkChainForDivergence checks a single chain for reth/geth divergence
+func checkChainForDivergence(
+	t systest.T,
+	chain system.L2Chain,
+	logger log.Logger,
+	testTimeout time.Duration,
+	blockCheckCount uint64,
+) {
+	// Create context with timeout
+	ctx, cancel := context.WithTimeout(t.Context(), testTimeout)
+	defer cancel()
+
+	// Find geth and reth nodes
+	gethNode, rethNode, err := findGethAndRethNodes(chain, logger)
+	if err != nil {
+		t.Skip(err.Error())
```

Comment:
Is it reasonable to skip here, given that not all chains will have both a geth and reth node?
Note that if we skip, the acceptance tests will fail.

Reply:
You could change `findGethAndRethNodes` so that it doesn't return an error if there aren't any reth nodes found. Instead you could add an extra condition here that just returns (i.e. test passes) if no reth nodes are found. Or do we explicitly want this acceptance test to skip+fail if there are no reth nodes?

Reply:
Thanks. I don't think that skip+fail makes sense as it seems completely reasonable for there to be no reth nodes.
I'll change this to how you suggested while we await for more feedback here.